python/tskit/trees.py

```diff
@@ -148,6 +148,10 @@ def __eq__(self, other):
             and self.metadata == other.metadata
         )
 
+    def __post_init__(self):
+        self.time = None
+        self.population = None
+
```

Comment:
Hmm, what about:
```
@property
def time(self):
    times = {self.node(u).time for u in self.nodes}
    if len(times) != 1:
         raise ValueError("This individuals does not have a well defined time")
    return times.pop()
```

Reply:
Only if you did ``ts.individual(k).time`` or ``[ind.time for ind in ts.individuals()]`` - which is all you can do I think?

This can't break any existing code because there's no place that we use these attributes in the library (which we should be *very* careful about doing, I agree)

Reply:
Sorry, I was being too hasty. I don't think we can do just this because (unless I'm wrong?) here `self` is an Individual object, not the TreeSequence, and so we can't go back and pull out the times from the nodes in a `@property`? Ah, but I could set `self.times = {self.node(u).time for u in nodes}` in `ts.individual(k)` and then do like so:
```
@property
def time(self):
  if len(self.times) == 1:
    return self.times[0]  # not right because it's a set
  else:
    raise ValueError(...)
```